hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -412,4 +417,86 @@ public JobConfig setInitialSnapshotName(@Nullable String initialSnapshotName) {
         this.initialSnapshotName = initialSnapshotName;
         return this;
     }
+
+    @Override
+    public int getFactoryId() {
+        return JetConfigDataSerializerHook.FACTORY_ID;
+    }
+
+    @Override
+    public int getId() {
+        return JetConfigDataSerializerHook.JOB_CONFIG;
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeUTF(name);
+        out.writeObject(processingGuarantee);
+        out.writeLong(snapshotIntervalMillis);
+        out.writeBoolean(autoScaling);
+        out.writeBoolean(splitBrainProtectionEnabled);
+        SerializationUtil.writeNullableList(resourceConfigs, out);
```

Comment:
we have built in serializer for list, why not just use writeObject?

Reply:
I've forgot that we have them in place